include/boost/sml.hpp

```diff
@@ -1587,14 +1587,6 @@ class sm {
     aux::get<sm_impl<TSM>>(sub_sms_).start(deps_, sub_sms_);
   }
   sm(aux::init, deps_t &deps) : deps_{deps}, sub_sms_{deps} { aux::get<sm_impl<TSM>>(sub_sms_).start(deps_, sub_sms_); }
```

Comment:
@maximilianriemensberger could you provide tests to cover this issue, please?

Reply:
I pushed tests for those removed constructors. I also tried to remove the `#if !defined(_MSC_VER)` around the move assignment test. I'd hope that removing those ctors and assignment operators might also fix the problem with MSVC but I can't test that locally.

Reply:
I think the tests are now working. It turns out that then variadic constructor of sm needs to be SFINAEd-away because it takes precedence over the copy constructor if an lvalue reference is passed (e.g. in the new copy ctor test).

Reply:
Please review and merge once checks are green. Thanks.

Reply:
Just for clarity on the status of the four new tests on master:
* `sm_move_assignment` test passes.
* `sm_copy_assignment` fails to compile on master
* `sm_copy_ctor` compiles but fails at runtime.
* `sm_move_ctor` compiles but fails at runtime.